week-3/todo-list/script.js

```diff
@@ -1,25 +1,96 @@
+//   event.preventDefault();
+//   // Write your code here... and remember to reset the input field to be blank after
+//   const input = document.querySelector("#input").value;
+//   const nweToDo = { task: input };
+//   // const li = document.createElement("li");
+//   // ul.appendChild(li);
+
+//   // li.textContent = input;
+// }
+
+// // Advanced challenge: Write a fucntion that checks the todos in the todo list and deletes the completed ones (we can check which ones are completed by seeing if they have the line-through styling applied or not).
+// function deleteAllCompletedTodos() {
+//   // Write your code here...
+// }
+
+
+
 function populateTodoList(todos) {
   let list = document.getElementById("todo-list");
-  // Write your code to create todo list elements with completed and delete buttons here, all todos should display inside the "todo-list" element.
+
+  todos.forEach((todo) => {
+    const li = document.createElement("li");
+    li.textContent = todo.task;
+
+    const span = document.createElement("span");
+    span.className = "badge bg-primary rounded-pill";
+
+    const checkIcon = document.createElement("i");
+    checkIcon.className = "fa fa-check";
+    checkIcon.setAttribute("aria-hidden", "true");
+    checkIcon.addEventListener("click", () => toggleCompleted(li));
+
+    const trashIcon = document.createElement("i");
+    trashIcon.className = "fa fa-trash";
+    trashIcon.setAttribute("aria-hidden", "true");
+    trashIcon.addEventListener("click", () => deleteTodo(li));
+
+    span.appendChild(checkIcon);
+    span.appendChild(trashIcon);
+
+    li.appendChild(span);
+    list.appendChild(li);
+  });
+}
+
+function toggleCompleted(todoElement) {
+  todoElement.classList.toggle("completed");
 }
 
-// These are the same todos that currently display in the HTML
-// You will want to remove the ones in the current HTML after you have created them using JavaScript
-let todos = [
-  { task: "Wash the dishes", completed: false },
-  { task: "Do the shopping", completed: false },
-];
+function deleteTodo(todoElement) {
+  todoElement.remove();
+}
+
+const button = document.querySelector("#add_todo");
+button.addEventListener("click", addNewTodo);
 
-populateTodoList(todos);
+const removeAllCompletedButton = document.querySelector("#remove-all-completed");
+removeAllCompletedButton.addEventListener("click", deleteAllCompletedTodos);
 
-// This function will take the value of the input field and add it as a new todo to the bottom of the todo list. These new todos will need the completed and delete buttons adding like normal.
 function addNewTodo(event) {
-  // The code below prevents the page from refreshing when we click the 'Add Todo' button.
   event.preventDefault();
-  // Write your code here... and remember to reset the input field to be blank after creating a todo!
+
+  const input = document.querySelector("#input").value;
+  const newToDo = { task: input, completed: false };
+  
+  populateTodoList([newToDo]);
```

Comment:
This works because you passing a new list of one item here.  Your `newToDo` really should be added to a list called `todos`.
Your solution is working for adding items to the list, but setting items complete or deleting items is not working (as far as I can see).